lib/mjit/compiler.rb

```diff
@@ -73,23 +73,6 @@ def compile_body(f, iseq, status)
       src << "#undef GET_SELF\n"
       src << "#define GET_SELF() cfp_self\n"
 
-      # Generate merged ivar guards first if needed
```

Comment:
I think you shouldn't have deleted the `merge_ivar_guards_p` optimization because checking `RB_TYPE_P(obj, T_OBJECT)` against self every time remains to be redundant. Now this place should now check only `RB_TYPE_P(GET_SELF(), T_OBJECT)` and the check should be removed from the following places when `merge_ivar_guards_p`.

Reply:
Sorry, didn't mean to remove this. Should be fixed now!